src/main/java/com/minecolonies/core/colony/ColonyManager.java

```diff
@@ -134,46 +139,54 @@ public void deleteColonyByDimension(final int id, final boolean canDestroy, fina
      */
     private void deleteColony(@Nullable final IColony iColony, final boolean canDestroy)
     {
-        if (!(iColony instanceof Colony))
+        if (!(iColony instanceof final Colony colony))
         {
             return;
         }
 
-        final Colony colony = (Colony) iColony;
         final int id = colony.getID();
         final Level world = colony.getWorld();
 
         if (world == null)
         {
-            Log.getLogger().warn("Deleting Colony " + id + " errored: World is Null");
+            Log.getLogger().warn("Deleting Colony {} errored: World is Null", id);
+            return;
+        }
+
+        final IColonyManagerCapability cap = world.getCapability(COLONY_MANAGER_CAP, null).resolve().orElse(null);
+        if (cap == null)
```

Comment:
why move this all the way up there?

Reply:
Most other methods have a fast-fail if the colony manager capability is missing. It's a bit strange if we first go and bother to remove all data from the colony, to then realize "oh we actually can't delete the colony".

Reply:
This is a "can't happen" thing anyway

Reply:
please undo re-org otherwise it's a pain to backport